build/mongodl.py

```diff
@@ -0,0 +1,665 @@
+            edition = dl['edition']
+            ar_url = dl['archive']['url']
+            db.execute(
+                r'''
+                INSERT INTO downloads (version_id, target, arch, edition, ar_url, data)
+                VALUES (?, ?, ?, ?, ?, ?)
+                ''',
+                (version_id, target, arch, edition, ar_url, json.dumps(dl)),
+            )
+            dl_id = db.lastrowid
+            for key, data in dl.items():
+                if 'url' not in data:
+                    continue
+                db.execute(
+                    r'''
+                    INSERT INTO components (key, download_id, data)
+                    VALUES (?, ?, ?)
+                    ''',
+                    (key, dl_id, json.dumps(data)),
+                )
+
+
+def get_dl_db():
+    caches = cache_dir()
+    caches.mkdir(exist_ok=True, parents=True)
+    db = sqlite3.connect(str(caches / 'downloads.db'), isolation_level=None)
+    db.executescript(r'''
+        CREATE TABLE IF NOT EXISTS meta (
+            etag TEXT,
+            last_modified TEXT
+        )
+    ''')
+    db.executescript(r'''
+        CREATE TABLE IF NOT EXISTS past_downloads (
+            url TEXT NOT NULL UNIQUE,
+            etag TEXT,
+            last_modified TEXT
+        )
+    ''')
+    changed, full_json = _download_file(
+        db, 'https://downloads.mongodb.org/full.json')
+    if not changed:
+        return db
+    with db:
+        print('Refreshing downloads manifest ...')
+        cur = db.cursor()
+        cur.execute("begin")
+        _import_json_data(cur, full_json)
+    return db
+
+
+def _print_list(db, version, target, arch, edition, component):
+    if version or target or arch or edition or component:
+        matching = db.execute(
+            r'''
+            SELECT version, target, arch, edition, key, components.data
+              FROM components,
+                   downloads USING(download_id),
+                   versions USING(version_id)
+            WHERE (:component IS NULL OR key=:component)
+              AND (:target IS NULL OR target=:target)
+              AND (:arch IS NULL OR arch=:arch)
+              AND (:edition IS NULL OR edition=:edition)
+              AND (:version IS NULL OR version=:version)
+            ''',
+            dict(version=version,
+                 target=target,
+                 arch=arch,
+                 edition=edition,
+                 component=component),
+        )
+        found_any = False
+        for version, target, arch, edition, comp_key, comp_data in matching:
+            found_any = True
+            print('Download: {}\n'
+                  ' Version: {}\n'
+                  '  Target: {}\n'
+                  '    Arch: {}\n'
+                  ' Edition: {}\n'
+                  '    Info: {}\n'.format(comp_key, version, target, arch,
+                                          edition, comp_data))
+    arches, targets, editions, versions, components = next(
```

Comment:
IIUC the intent is to not print all unique values if there are any matches.
```suggestion
    if found_any:
        return
    arches, targets, editions, versions, components = next(
```